### PR TITLE
[POAE7-2746] Upgrade Optimization PassManager from Legacy PM to New PM

### PR DESCRIPTION
**Pull request by bigPYJ1151**
### What changes were proposed in this pull request?

Upgrade Optimization PassManager from Legacy PM to New PM

### Why are the changes needed?

JITLib Optimization

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UTs

### Which label does this PR belong to?

Infra

### CODE REVIEW

File: src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITModule.cpp [hunk 1]
```diff
@@ -142,54 +155,62 @@ void LLVMJITModule::finish() {
...
+                                      function_analysis_mgr,
+                                      cgscc_analysis_mgr,
+                                      module_analysis_mgr);
+
+    llvm::ModulePassManager module_pass_mgr;
+    llvm::FunctionPassManager function_pass_mgr;
+    llvm::LoopPassManager loop_pass_mgr;
+
+    module_pass_mgr.addPass(
+        llvm::AlwaysInlinerPass(false));  // Inline all functions labeled as always_inline
```
Comment (Ma-Jian1) [inline src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITModule.cpp:189]: `false` means inline always?

Comment (bigPYJ1151) [reply src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITModule.cpp:189]: image 

Nope

**Review by Ma-Jian1 — APPROVED** [review]

File: src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITModule.cpp [hunk 2]
```diff
@@ -158,54 +171,62 @@ void LLVMJITModule::finish(const std::string& main_func) {
...
+    function_pass_mgr.addPass(
+        llvm::SimplifyCFGPass());  // remove load/stores in PHIs if instructions can be
+                                   // accessed directly post thread jumps
+    function_pass_mgr.addPass(llvm::NewGVNPass());
+    function_pass_mgr.addPass(llvm::DSEPass());
+
+    function_pass_mgr.addPass(llvm::InstCombinePass());
+    function_pass_mgr.addPass(llvm::PromotePass());
+
+    loop_pass_mgr.addPass(llvm::LICMPass());
```
Comment (xieqi) [inline src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITModule.cpp:220]: Do we need to add any loop vectorizer related pass?

**Review by xieqi — APPROVED** [review]

File: src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITModule.cpp (hunk 2 above)
Comment (bigPYJ1151) [reply src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITModule.cpp:220]: Yes

**Pull request merged**